### DISCUSSION TITLE
Design is only counting 1000 rows

### DISCUSSION THREAD

**Discussion by mojoaar** [Budibase/budibase#6006] [Help] [answered]
Hi all,

I'm new to Budibase and trying to do a basic cmdb. On the front there's some statistics displayed.

![1](https://user-images.githubusercontent.com/21560785/169558164-eaa3ee08-8a0e-4b36-a345-d4d9d700087f.png)

When I'm in the design section I can only get it to display up to 1000 rows with Length ( {{ All Servers.Rows Length }} ).

![2](https://user-images.githubusercontent.com/21560785/169558440-52b545ed-4124-4d50-bfc3-078ee30ad8ae.png)

If I do a View in Data and do a calculation I get the correct count of rows (2316).

![3](https://user-images.githubusercontent.com/21560785/169558132-c58bf999-42f0-4ffb-9d35-edebe5744f88.png)

Can anyone tell me what I am doing wrong in the Display section?

Kind Regards
Morten

**Comment by melohagan** [comment, marked as answer]
Hi @mojoaar 

Only 1000 rows can be returned per query / load for internal tables.

**Reply by mojoaar** [reply]
Thanks @melohagan.

Any way of working around the limitation - JavaScript maybe?

**Reply by melohagan** [reply]
The internal database uses Lucene for filtering and querying data. This makes lifting the 1000 row limit difficult. 
What you could do is paginate your data provider, with max 1000 rows per page.

There is no such limit on external datasources when self hosting - so that may be an option. 
If you're processing large amounts of data, or building graphs and dashboards, then you're better using an external DB.

**Reply by mike12345567** [reply]
Hi @mojoaar another option if you don't need dynamic data filtering is to use a view on your internal table, as these statically filter the data upfront they can return much larger amounts of data. In the Cloud these have limited functionality right now but are fully capable of returning as much data as you need in self host. You can find documentation about views [here](https://docs.budibase.com/docs/views).

**Reply by mojoaar** [reply]
Thanks for both answers - think I'll look at using an external db source.
